### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -138,7 +138,7 @@ As always, we appreciate contributions of bug reports, code fixes,
 and sample submission from the ClamAV community members:
 
 Sebastian Andrzej Siewior
-Minzhaun Gong
+Minzhuan Gong
```
Comment (jsoref) [inline README]: https://www.linkedin.com/in/minzhuan-gong-8b73651b

File: contrib/old-clamav-milter/Makefile.in [hunk 2]
```diff
@@ -522,7 +522,7 @@ ID: $(HEADERS) $(SOURCES) $(LISP) $(TAGS_FILES)
 	unique=`for i in $$list; do \
 	    if test -f "$$i"; then echo $$i; else echo $(srcdir)/$$i; fi; \
 	  done | \
-	  $(AWK) '{ files[$$0] = 1; nonemtpy = 1; } \
+	  $(AWK) '{ files[$$0] = 1; nonempty = 1; } \
```
Comment (jsoref) [inline contrib/old-clamav-milter/Makefile.in]: This is an actual code bug

File: contrib/old-clamav-milter/clamav-milter.c [hunk 3]
```diff
@@ -675,7 +675,7 @@ help(void)
 	puts(_("\t--headers\t\t-H\tInclude original message headers in the report."));
 	puts(_("\t--ignore IPaddr\t\t-I IPaddr\tAdd IPaddr to LAN IP list (see --local)."));
 	puts(_("\t--local\t\t\t-l\tScan messages sent from machines on our LAN."));
-	puts(_("\t--max-childen\t\t-m\tMaximum number of concurrent scans."));
+	puts(_("\t--max-children\t\t-m\tMaximum number of concurrent scans."));
```
Comment (jsoref) [inline contrib/old-clamav-milter/clamav-milter.c]: The output didn't match the parser

File: libclamav/bytecode.c [hunk 4]
```diff
@@ -820,7 +820,7 @@ static int parseTypes(struct cli_bc *bc, unsigned char *buffer)
 }
 
 /* checks whether the type described by tid is the same as the one described by
- * expectty. */
+ * apitid. */
```
Comment (jsoref) [inline libclamav/bytecode.c]: I know this change looks drastic, but it matches the code.

File: libclamav/bytecode.c [hunk 5]
```diff
@@ -3132,7 +3132,7 @@ static void cli_bytetype_helper(const struct cli_bc *bc, unsigned tid)
 
     i = tid - 65;
     if (i >= bc->num_types) {
-        printf("invaltype");
+        printf("invalid type");
```
Comment (jsoref) [inline libclamav/bytecode.c]: I hope this change is acceptable.

File: libclamav/mbox.c [hunk 6]
```diff
@@ -1037,38 +1037,38 @@ parseEmailHeader(message *m, const char *line, const table_t *rfc821)
 #ifdef CL_THREAD_SAFE
 	char *strptr;
 #endif
-	const char *separater;
-	char *cmd, *copy, tokenseparater[2];
+	const char *separator;
+	char *cmd, *copy, tokenseparator[2];
```
Comment (jsoref) [inline libclamav/mbox.c]: This changes local variables, it should be safe.

File: libclamav/mbox.c [hunk 7]
```diff
@@ -3754,7 +3754,7 @@ exportBounceMessage(mbox_ctx *mctx, text *start)
 	 * helps a bit.
 	 *
 	 * messageAddLine
-	 * optimisation could help here, but needs
+	 * optimization could help here, but needs
```
Comment (jsoref) [inline libclamav/mbox.c]: en-US is dominant in this project.

File: libclamav/others.c [hunk 8]
```diff
@@ -86,7 +86,7 @@ int (*cli_unrar_extract_next_prepare)(unrar_state_t *state, const char *dirname)
 int (*cli_unrar_extract_next)(unrar_state_t *state, const char *dirname);
 void (*cli_unrar_close)(unrar_state_t *state);
 int have_rar = 0;
-static int is_rar_initd = 0;
+static int is_rar_inited = 0;
```
Comment (jsoref) [inline libclamav/others.c]: This is a local variable, so using the more common spelling seems reasonable.

File: libclamav/sis.c [hunk 9]
```diff
@@ -407,13 +407,13 @@ static int real_scansis(cli_ctx *ctx, const char *tmpd) {
 	sftype = "sub sis";
 	break;
       case FTcontsis:
-	sftype = "conatiner sis";
+	sftype = "container sis";
```
Comment (jsoref) [inline libclamav/sis.c]: I hope these changes aren't problematic.

File: libclamav/tiff.c [hunk 10]
```diff
@@ -83,7 +83,7 @@ int cli_parsetiff(cli_ctx *ctx)
 
         cli_dbgmsg("cli_parsetiff: IFD %u declared %u directory entries\n", ifd_count, num_entries);
 
-        /* tranverse IFD entries */
+        /* transverse IFD entries */
```
Comment (jsoref) [inline libclamav/tiff.c]: This comes from the tiff spec....

File: libclamav/tiff.c [hunk 11]
```diff
@@ -101,7 +101,7 @@ int cli_parsetiff(cli_ctx *ctx)
             case 1: /* BYTE */
                 value_size *= 1;
                 break;
-            case 2: /* ACSII */
+            case 2: /* ASCII */
```
Comment (jsoref) [inline libclamav/tiff.c]: I hope this is right....

File: m4/libtool.m4 [hunk 12]
```diff
@@ -651,7 +651,7 @@ configured by $[0], generated by m4_PACKAGE_STRING.
 
 Copyright (C) 2011 Free Software Foundation, Inc.
 This config.lt script is free software; the Free Software Foundation
-gives unlimited permision to copy, distribute and modify it."
+gives unlimited permission to copy, distribute and modify it."
```
Comment (jsoref) [inline m4/libtool.m4]: This is license text, if you want to keep the typo, that's understandable.

**Comment by kevlin2** [comment]
The changes to configure are unusable because configure is generated from various source files. likely most of the spelling changes can be ported over to the corresponding configure.ac.

Besides the conflicts and configure changes, the reset is good to merge as-is.

**Comment by kevlin2** [comment]
The spelling changes as well as a number of changes added by me have been submitted for internal review. 
Pending review, the changes are likely to be included in the 0.100 release and at latest the subsequent release.

**Comment by micah-at-talos** [comment]
The spelling changes were included with some manual adjustment in commit https://github.com/Cisco-Talos/clamav-devel/commit/1dfeefc0fdf3277e046e9a4dd04d41ce32f1bb2e

**Pull request closed without merging**